app/models/post.rb

```diff
@@ -1,2 +1,17 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  attribute :Post_counter, default: 0
+
+  # Increment the Post_counter for the author
+  def increment_post_counter_for_user
+    author.increment(:Post_counter, 1)
+  end
+
```

Comment:
- [ ] **[OPTIONAL]** It would be a good idea to have an active record callback for this method. You can add an `after_save` callback so that when the post is created it triggers the method on its own.

- [ ] **[OPTIONAL]** You can also make the method private.

  ```
  after_save :increment_post_counter_for_user
    author.increment!(:Post_counter)
  end
  ```